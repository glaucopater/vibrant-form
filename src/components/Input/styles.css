:root {
    --text-shadows: 2px 2px 4px rgba(0, 0, 0, 0.2);
    --box-shadow: 4px 4px 60px rgba(0, 0, 0, 0.2);
    --custom-border: 1px solid rgba(255, 255, 255, 0.3);
    --validation-color: #eaf120;
    --white: #f1f1f1;
}

.vibrantFormInput {
    display: flex;
    flex-direction: column;
    width: 100%;
}

.vibrantFormLabel {
    color: var(--white);
    font-size: 16px;
    opacity: 0.7;
    text-shadow: var(--text-shadow);
    align-self: flex-start;
    padding: 4px 0;
    text-transform: capitalize;
}

.vibrantFormInputText {
    background: transparent;
    padding: 1em;
    margin-bottom: 2em;
    border: none;
    border-left: var(--custom-border);
    border-top: var(--custom-border);
    border-radius: 20px;
    -webkit-backdrop-filter: blur(5px);
    backdrop-filter: blur(5px);
    box-shadow: var(--box-shadow);
    color: var(--white);
    font-weight: 500;
    transition: all 0.2s ease-in-out;
    text-shadow: var(--text-shadow);
    opacity: 0.7;
    outline: none;
}

.vibrantFormInputText:hover {
    background: rgba(255, 255, 255, 0.1);
    box-shadow: var(--box-shadow);
}

.vibrantFormInputText[type="text"]:focus {
    background: rgba(255, 255, 255, 0.1);
    box-shadow: var(--box-shadow);
    opacity: 1;
}

.vibrantFormInputText[type="button"] {
    margin-top: 10px;
    width: 150px;
    font-size: 1rem;
}

.vibrantFormInputText[type="button"]:hover {
    cursor: pointer;
}

.vibrantFormInputText[type="button"]:active {
    background: rgba(255, 255, 255, 0.2);
}


::-moz-placeholder {
    font-weight: 400;
    color: var(--white);
    text-shadow: var(--text-shadow);
}

:-ms-input-placeholder {
    font-weight: 400;
    color: var(--white);
    text-shadow: var(--text-shadow);
}

::placeholder {
    font-weight: 400;
    color: var(--white);
    text-shadow: var(--text-shadow);
    font-style: italic;
}

.vibrantFormLabelContainer {
    display: flex;
    place-items: center;
    justify-content: space-between;
    width: 100%;
}

.validationMessage {
    color: var(--validation-color);
    font-size: 16px;
    opacity: 0.7;
    text-shadow: var(--text-shadow);
    align-self: flex-start;
    padding: 2px 0;
}