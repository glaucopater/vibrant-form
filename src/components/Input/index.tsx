
import React from "react";
import { dictionary } from "../../dictionary";
import { getClassNameProps } from "../../helpers";
import { InputPropsType, ValidationErrorType } from "../../types";
import "./styles.css";

const ValidationMessage: React.FC<{ errors: ValidationErrorType[], withTheme?: boolean }> = ({ errors, withTheme }) => {
    const message = errors.length > 0 ? errors[0].message : dictionary.validated;
    const classNameProps = withTheme ? { className: "validationMessage" } : null;
    return <span {...classNameProps} > {message}</ span >;
}

const Input: React.FC<InputPropsType> = ({ name, value, type, isRequired, placeholder, pattern, nativeValidation = false, errors, isValidated, withTheme }) => {

    const nativeValidationProps = {
        ...(nativeValidation && {
            required: isRequired,
            "aria-required": isRequired,
            pattern
        })
    }

    return (
        <div {...getClassNameProps("vibrantFormInput", withTheme)} >
            <div {...getClassNameProps("vibrantFormLabelContainer", withTheme)} >
                <label {...getClassNameProps("vibrantFormLabel", withTheme)} htmlFor={`${name}InputText`}>
                    {name}
                </label>
                {isValidated && <ValidationMessage errors={errors} withTheme={withTheme} />}
            </div>
            <input {...getClassNameProps("vibrantFormInputText", withTheme)}
                id={`${name}InputText`}
                name={name}
                type={type}
                defaultValue={value}
                placeholder={placeholder || ""}
                {...nativeValidationProps}
            />
        </div>
    );
}

export default Input;