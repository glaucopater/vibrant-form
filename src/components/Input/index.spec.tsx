import Input from ".";

describe("Input", () => {
    it("should render the empty component", () => {
        expect(Input).toMatchSnapshot();
    })
})
