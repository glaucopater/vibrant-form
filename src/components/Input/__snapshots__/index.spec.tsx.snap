// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render the empty component 1`] = `[Function]`;
